### DISCUSSION TITLE
Are Text Input Components not supported?

### DISCUSSION THREAD

**Discussion by Joshua-Wright76** [discordjs/discord.js#7651] [Q&A] [answered]
In the [official API documentation](https://discord.com/developers/docs/interactions/message-components#component-object-component-types), there are four component types: Action rows, Selects, Buttons, and Text Inputs. I haven't been able to figure out how to create Text Input components, is there any way of making them, or are they just not supported right now?

**Comment by monbrey** [comment, marked as answer]
TextInputs are only available in Modals. They are otherwise not a "message" component

Modals will be available in discord.js 13.7 and 14

**Reply by SwatDoge** [reply]
Is this still the case today?

**Reply by almostSouji** [reply]
yes
